addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/JsFunctionScanRule.java

```diff
@@ -57,6 +59,8 @@
     public static final String JS_FUNCTION_PAYLOAD_CATEGORY = "JS-Function";
 
     private static Supplier<Iterable<String>> payloadProvider = DEFAULT_PAYLOAD_PROVIDER;
+    private static final Map<String, String> ALERT_TAGS =
+            CommonAlertTag.toMap(CommonAlertTag.OWASP_2021_A04_INSECURE_DESIGN);
```

Comment:
I couldn't see a good match for 2017.

Reply:
Me neither. Insecure design feels like a bit of a stretch too but I cant see anything better :/